core/domain/stats_jobs_one_off.py

```diff
@@ -303,6 +315,25 @@ def reduce(exp_id, values):
 
     @classmethod
     def prepare_reduce(cls, exp_id, values):
+        """Runs the reduce step to extract aggregate exploration statistics from
+        the dicts passed in by the mapping function.
+
+        Args:
+            exp_id: str. The exploration id.
+            values: list(dict(str, str)). List of dicts of events for preparing
+                the reduce.
+
+        Returns:
+            tuple(list(dict), list(ExplorationStats), list(str)). 3-tuple where:
+            - The first element is the list of dicts of different version-wise
+                ExplorationStats. Each dict contains the following key-value
+                pairs:
```

Comment:
I think this needs the list of key-value pairs (and their types/descriptions).

Reply:
For description about the dict: The keys are dependent on version numbers like: num_starts_v2, num_completions_v2, ... etc. Also, there are around 10 keys like this and it doesn't make much sense to me. Shall I write them here and how to handle versions in keys?